docs/DeveloperGuide.md

```diff
@@ -0,0 +1,568 @@
+`UiPart` class.
+
+The `UI` component uses JavaFx UI framework. The layout of these UI
+parts are defined in matching `.fxml` files that are in the
+`src/main/resources/view` folder. For example, the layout of the
+[`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java)
+is specified in
+[`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
+
+The `UI` component,
+
+  - Executes user commands using the `Logic` component.
+
+  - Listens for changes to `Model` data so that the UI can be updated
+    with the modified data.
+
+### Logic component
+
+![Structure of the Logic Component](images/LogicClassDiagram.png)
+
+**API** :
+[`Logic.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/logic/Logic.java)
+
+1.  `Logic` uses the `AddressBookParser` class to parse the user
+    command.
+
+2.  This results in a `Command` object which is executed by the
+    `LogicManager`.
+
+3.  The command execution can affect the `Model` (e.g. adding a person).
+
+4.  The result of the command execution is encapsulated as a
+    `CommandResult` object which is passed back to the `Ui`.
+
+5.  In addition, the `CommandResult` object can also instruct the `Ui`
+    to perform certain actions, such as displaying help to the user.
+
+Given below is the Sequence Diagram for interactions within the `Logic`
+component for the `execute("delete 1")` API call.
+
+![Interactions Inside the Logic Component for the `delete 1`
+Command](images/DeleteSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: <b>Note: </b>The lifeline for `DeleteCommandParser` should end at the
+destroy marker (X) but due to a limitation of PlantUML, the lifeline
+reaches the end of diagram.
+</div>
+
+### Model component
+
+![Structure of the Model Component](images/ModelClassDiagram.png)
+
+**API** :
+[`Model.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/model/Model.java)
+
+The `Model`,
+
+  - stores a `UserPref` object that represents the user’s preferences.
+
+  - stores the Address Book data.
+
+  - exposes an unmodifiable `ObservableList<Person>` that can be
+    'observed' e.g. the UI can be bound to this list so that the UI
+    automatically updates when the data in the list change.
+
+  - does not depend on any of the other three components.
+
+
+<div markdown="span" class="alert alert-info">:information_source: <b>Note: </b>As a more OOP model, we can store a `Tag` list in `Address Book`, which
+`Person` can reference. This would allow `Address Book` to only require
+one `Tag` object per unique `Tag`, instead of each `Person` needing
+their own `Tag` object. An example of how such a model may look like is
+given below.
+
+![BetterModelClassDiagram](images/BetterModelClassDiagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/1673303/79063516-cf3cca00-7cd4-11ea-93c9-56c9f5861a0a.png)
Line break before diagram?

Reply:
Added a <br> tag because normal markdown breaks don't seem to work.